### PR TITLE
java11 compatibility

### LINKED ISSUE #1232: Java 11 EA Build: Compile Errors: ambiguous reference both <T>assertThat(T) and method assertThat(java.lang.CharSequence) match
Just for your information.

When running `mvn clean test` using:

```
java version "11-ea" 2018-09-25
Java(TM) SE Runtime Environment 18.9 (build 11-ea+10)
Java HotSpot(TM) 64-Bit Server VM 18.9 (build 11-ea+10, mixed mode)
```

I get these compile errors:

```
[ERROR] /C:/Users/Redhawk/dev/assertj-core/src/test/java/org/assertj/core/api/AutoCloseableBDDSoftAssertionsTest.java:[62,13] reference to then is ambiguous
  both method <T>then(T) in org.assertj.core.api.Java6AbstractBDDSoftAssertions and method then(java.lang.CharSequence) in org.assertj.core.api.Java6AbstractBDDSoftAssertions match
[ERROR] /C:/Users/Redhawk/dev/assertj-core/src/test/java/org/assertj/core/api/SoftAssertionsTest.java:[219,13] reference to assertThat is ambiguous
  both method <T>assertThat(T) in org.assertj.core.api.Java6AbstractStandardSoftAssertions and method assertThat(java.lang.CharSequence) in org.assertj.core.api.Java6AbstractStandardSoftAssertions match
[ERROR] /C:/Users/Redhawk/dev/assertj-core/src/test/java/org/assertj/core/api/BDDSoftAssertionsTest.java:[216,13] reference to then is ambiguous
  both method <T>then(T) in org.assertj.core.api.Java6AbstractBDDSoftAssertions and method then(java.lang.CharSequence) in org.assertj.core.api.Java6AbstractBDDSoftAssertions match
[ERROR] /C:/Users/Redhawk/dev/assertj-core/src/test/java/org/assertj/core/perf/SoftAssertionsPerfTest.java:[192,13] reference to assertThat is ambiguous
  both method <T>assertThat(T) in org.assertj.core.api.Java6AbstractStandardSoftAssertions and method assertThat(java.lang.CharSequence) in org.assertj.core.api.Java6AbstractStandardSoftAssertions match
[ERROR] /C:/Users/Redhawk/dev/assertj-core/src/test/java/org/assertj/core/api/AutoCloseableSoftAssertionsTest.java:[71,13] reference to assertThat is ambiguous
  both method <T>assertThat(T) in org.assertj.core.api.Java6AbstractStandardSoftAssertions and method assertThat(java.lang.CharSequence) in org.assertj.core.api.Java6AbstractStandardSoftAssertions match
[ERROR] /C:/Users/Redhawk/dev/assertj-core/src/test/java/org/assertj/core/api/Assertions_assertThat_with_CharSequence_Test.java:[35,61] reference to assertThat is ambiguous
  both method assertThat(java.lang.CharSequence) in org.assertj.core.api.Assertions and method <T>assertThat(T) in org.assertj.core.api.Assertions match
[ERROR] /C:/Users/Redhawk/dev/assertj-core/src/test/java/org/assertj/core/api/Assertions_assertThat_with_CharSequence_Test.java:[41,61] reference to assertThat is ambiguous
  both method assertThat(java.lang.CharSequence) in org.assertj.core.api.Assertions and method <T>assertThat(T) in org.assertj.core.api.Assertions match
```

### PR DESCRIPTION
**Pull request by epeee**
#### Check List:
* Fixes  #1232
* Unit tests : YES / NO / NA
* Javadoc with a code example (API only) : YES / NO / NA

Should we add some additional test too or are we fine with the ones we are having in place?
Additionally, should we merge  #1234 first and I do a rebase afterwards and make a change to no longer allow failing java11 builds?

### CODE REVIEW

File: src/main/java/org/assertj/core/api/Assertions.java [hunk 1]
```diff
@@ -2565,6 +2565,29 @@ public static void useDefaultDateFormatsOnly() {
...
+  /**
+   * Creates a new instance of <code>{@link CharSequenceAssert}from a {@link StringBuilder}</code>.
+   *
+   * @param actual the actual value.
+   * @return the created assertion object.
+   * @since 3.11.0
+   */
+  @CheckReturnValue
+  public static AbstractCharSequenceAssert<?, ? extends CharSequence> assertThat(StringBuilder actual) {
+    return new CharSequenceAssert(actual);
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/api/Assertions.java]: Should be in `AssertionsForClassTypes.assertThat(actual);` the current code being moved to `AssertionsForClassTypes` (as done for `assertThat(String actual)`)

File: src/main/java/org/assertj/core/api/Assertions.java [hunk 2]
```diff
@@ -2565,6 +2565,29 @@ public static void useDefaultDateFormatsOnly() {
...
+  /**
+   * Creates a new instance of <code>{@link CharSequenceAssert}from a {@link StringBuffer}</code>.
+   *
+   * @param actual the actual value.
+   * @return the created assertion object.
+   * @since 3.11.0
+   */
+  @CheckReturnValue
+  public static AbstractCharSequenceAssert<?, ? extends CharSequence> assertThat(StringBuffer actual) {
+    return new CharSequenceAssert(actual);
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/api/Assertions.java]: Should be in `AssertionsForClassTypes.assertThat(actual);` the current code being moved to `AssertionsForClassTypes` (as done for `assertThat(String actual)`)

**Comment by joel-costigliola** [comment]
Nice the build passes on JDK11 :smiley:  (on my local).

Two tests similar to[ `Assertions_assertThat_with_String_Test`](https://github.com/joel-costigliola/assertj-core/blob/cc178abc5871bc34525113550373ffd90e303034/src/test/java/org/assertj/core/api/Assertions_assertThat_with_String_Test.java) should be added. 

Good point for the build, I'll take care of #1234 once this PR is updated and ready to be merged.

**Comment by epeee** [comment]
Thanks for your feedback!
I've updated the PR accordingly.

**Comment by joel-costigliola** [comment]
Integrated thanks @epeee!

**Pull request merged**